neptune_optuna/impl/__init__.py

```diff
@@ -144,342 +175,484 @@ def __init__(self,
+                log_plot_pareto_front=self._log_plot_pareto_front,
+                log_plot_slice=self._log_plot_slice,
+                log_plot_optimization_history=self._log_plot_optimization_history,
+                log_plot_intermediate_values=self._log_plot_intermediate_values,
+                namespaces=self._namespaces,
+            )
 
     def _log_study(self, study, trial):
         if self._should_log_study(trial):
             _log_study(self.run, study)
 
-    def _should_log_plots(self, study: optuna.study.Study, trial: optuna.trial.FrozenTrial):
+    def _should_log_plots(
+        self, study: optuna.study.Study, trial: optuna.trial.FrozenTrial
+    ):
         if not len(study.get_trials(states=(optuna.trial.TrialState.COMPLETE,))):
             return False
-        elif self._plots_update_freq == 'never':
+        elif self._plots_update_freq == "never":
             return False
         else:
             if trial._trial_id % self._plots_update_freq == 0:
                 return True
         return False
 
     def _should_log_study(self, trial: optuna.trial.FrozenTrial):
-        if self._study_update_freq == 'never':
+        if self._study_update_freq == "never":
             return False
-        else:
-            if trial._trial_id % self._study_update_freq == 0:
-                return True
+        if trial._trial_id % self._study_update_freq == 0:
+            return True
         return False
 
 
-def log_study_metadata(study: optuna.Study,
-                       run: neptune.Run,
-                       base_namespace='',
-                       log_plots=True,
-                       log_study=True,
-                       log_all_trials=True,
-                       log_distributions=True,
-                       visualization_backend='plotly',
-                       log_plot_contour=True,
-                       log_plot_edf=True,
-                       log_plot_parallel_coordinate=True,
-                       log_plot_param_importances=True,
-                       log_plot_pareto_front=True,
-                       log_plot_slice=True,
-                       log_plot_intermediate_values=True,
-                       log_plot_optimization_history=True):
-    """A function that logs the metadata from Optuna Study to Neptune.
+def _log_best_trials(run, study: optuna.Study, namespaces):
+    if study._is_multi_objective():
+        _log_trials(
+            run,
+            study,
+            trials=study.best_trials,
+            namespaces=namespaces,
+            best=True,
+        )
+    else:
+        _log_single_trial(
+            run,
+            study,
+            trial=study.best_trial,
+            namespaces=namespaces,
+            best=True,
+        )
+
+
+def _get_namespaces(
+    study: optuna.Study, target_names: Optional[List[str]] = None
+) -> Union[List[str], str]:
+
+    if study._is_multi_objective():
```

Comment:
`study._is_multi_objective` could be changed in future versions of Optuna (it's internal attribute) or removed. Let's made our custom  function:
```python
def _is_multi_objective(study: optuna.Study) -> bool:
    return len(study.directions) > 1
```
I'm not testing it so take a look but that's how it was implemented in Optuna itself.

Reply:
Ok, so you would prefer to create this custom code.

Got it, consider it done by tomorrow ✅🫡

Reply:
Done ✅